### ISSUE TITLE
Data table extensions: header

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-cloud-cognitive#1608]
# Data table extensions: header

## Brief description

This issue will address the header data table extension.

## Details

[Link to Design](https://pages.github.ibm.com/cdai-design/pal/components/data-table/headers)
### Links to other materials

e.g. sketch files, icons, images, prototypes

## Design Maintainer(s)

- Marion Brülls

## Tasks

Before starting work on this epic, please review and complete the following.

- [ ] Initial review of design/existing code

### Working in Carbon for IBM Products package

- [ ] Have you reviewed our
      [contribution guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/.github/CONTRIBUTING.md)?
- [ ] Have you noted our
      [code guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/docs/CODE_GUIDELINES.md)?
- [ ] Did you use the CLI to scaffold your new component?

**Comment by elycheea** [comment]
Closed as duplicate of the newer #1931
